### ISSUE TITLE
Converting sparse IndexedSlices to a dense Tensor of unknown shape. This may consume a large amount of memory

### ISSUE THREAD

**Issue by linsocurry** [matterport/Mask_RCNN#347]
UserWarning: Converting sparse IndexedSlices to a dense Tensor of unknown shape. 
This may consume a large amount of memory.
"Converting sparse IndexedSlices to a dense Tensor of unknown shape. "

Hi. I use your code to train a mask-rcnn, but the training is ridiculous slow.  I google it and find someone meet the same problem https://stackoverflow.com/questions/35892412/tensorflow-dense-gradient-explanation, bu i still don't know how to fixed these problem, any advises?

**Comment by wangxm345566462** [comment]
same question

**Comment by kalanka29** [comment]
Can't  this warning caused regarding the dimensions of the images...I have used the mask rcnn and my training set is with larger resolution images.I am getting this warning and I think it is because of my dataset..plz help

**Comment by waleedka** [comment]
You can safely ignore this warning. It's a preemptive warning from TensorFlow when it cannot be certain of the size of the generated tensor.

**Comment by Jingyixiong** [comment]
Have someone already fixed this problem? I think it might because of the using of tf.gather() which transfers a sparse indexedslice to a dense tensor, it terribly consumed the graphic storage. Could anyone give a solution for this situation?
